libcloudforensics/providers/azure/internal/account.py

```diff
@@ -217,3 +262,213 @@ def CreateDiskFromSnapshot(
+
+    This is useful if e.g. one wants to make a copy of a disk in a separate
+    Azure account. This method will create a temporary Azure Storage account
+    within the destination account, import the snapshot from a downloadable
+    link (the source account needs to share the snapshot through a SAS link)
+    and then create a disk from the VHD file saved in storage. The Azure
+    storage account is then deleted.
+
+    Args:
+      snapshot (AZSnapshot): Source snapshot to use.
+      snapshot_uri (str): The URI of the snapshot to copy.
+      region (str): Optional. The region in which to create the disk. If not
+          provided, the disk will be created in the default_region associated to
+          the AZAccount object.
+      disk_name (str): Optional. String to use as new disk name.
+      disk_name_prefix (str): Optional. String to prefix the disk name with.
+      disk_type (str): Optional. The sku name for the disk to create. Can be
+          Standard_LRS, Premium_LRS, StandardSSD_LRS, or UltraSSD_LRS.
+          Default is Standard_LRS.
+
+    Returns:
+      AZDisk: Azure Compute Disk.
+
+    Raises:
+      RuntimeError: If the disk could not be created.
+    """
+
+    if not region:
+      region = self.default_region
+
+    # Create a temporary Azure account storage to import the snapshot
+    storage_account_name = hashlib.sha1(
+        snapshot.resource_id.encode('utf-8')).hexdigest()[:23]
+    storage_account_url = 'https://{0:s}.blob.core.windows.net'.format(
+        storage_account_name)
+    storage_account_id, storage_account_access_key = self._CreateStorageAccount(
+        storage_account_name, region=region)
+    blob_service_client = blob.BlobServiceClient(
+        account_url=storage_account_url, credential=storage_account_access_key)
+
+    # Create a container within the Storage to receive the imported snapshot
+    container_name = storage_account_name + '-container'
+    snapshot_vhd_name = snapshot.name + '.vhd'
+    container_client = blob_service_client.get_container_client(container_name)
+    try:
+      logger.info('Creating blob container {0:s}'.format(container_name))
+      container_client.create_container()
+      logger.info('Blob container {0:s} successfully created'.format(
+          container_name))
+    except exceptions.ResourceExistsError:
+      # The container already exists, so we can re-use it
+      logger.warning('Reusing existing container: {0:s}'.format(container_name))
+
+    # Download the snapshot from the URI to the storage
+    copied_blob = blob_service_client.get_blob_client(
+        container_name, snapshot_vhd_name)
+    logger.info('Importing snapshot to container from URI {0:s}. '
+                'Depending on the size of the snapshot, this process is going '
+                'to take a while.'.format(snapshot_uri))
+    copied_blob.start_copy_from_url(snapshot_uri)
+    while copied_blob.get_blob_properties().copy.status != 'success':
```

Comment:
I don't know what other status can be returned, but if it returns eg 'error' or something this will loop and loop and loop.

Reply:
Yeah, we pretty much have this problem with every Azure call. The thing is we have no way to predict how long these operations will take (since the size of the disk to copy will greatly influence the time spent looping there). We talked with @tomchop in one of the previous PRs about having a `MAX_RETRY` control variable, but we ended up not using it (for the reason mentioned previously, i.e. we can't decide on a good value for this variable without knowing the use case). Users can just quit the program if the feel like it's hanging. The call to the logger should provide them with at least a sense of "things are happening on Azure, let's wait"

Reply:
It would be pending when busy but can't we raise an error with the description when the status is 'failed'?
See https://docs.microsoft.com/en-us/rest/api/storageservices/get-blob-properties

Reply:
I see, added safe guard